src/test/java/com/bobocode/hoverla/bring/context/ApplicationContextImplTest.java

```diff
@@ -0,0 +1,152 @@
+package com.bobocode.hoverla.bring.context;
+
+import com.bobocode.hoverla.bring.exception.NoSuchBeanException;
+import com.bobocode.hoverla.bring.exception.NoUniqueBeanException;
+import com.google.common.collect.Table;
+import org.junit.jupiter.api.*;
+import org.mockito.Mockito;
+
+import java.lang.reflect.Field;
+import java.math.BigDecimal;
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@TestInstance(value = TestInstance.Lifecycle.PER_CLASS)
+public class ApplicationContextImplTest {
+
+    private ApplicationContext applicationContext;
+    private BeanScanner beanScannerOne;
+    private BeanScanner beanScannerTwo;
+    private BeanValidator validator;
+    private BeanInitializer initializer;
+
+    @BeforeAll
+    private void init() {
+        beanScannerOne = Mockito.mock(BeanScanner.class);
+        beanScannerTwo = Mockito.mock(BeanScanner.class);
+        List<BeanScanner> beanScannerList = Arrays.asList(beanScannerOne, beanScannerTwo);
+
+        validator = Mockito.mock(BeanValidator.class);
+
+        initializer = Mockito.mock(BeanInitializer.class);
+
+        BeanDefinition beanDefinitionStringOne = Mockito.mock(BeanDefinition.class);
+        BeanDefinition beanDefinitionStringTwo = Mockito.mock(BeanDefinition.class);
+        BeanDefinition beanDefinitionInteger = Mockito.mock(BeanDefinition.class);
+
+        when(beanScannerOne.scan()).thenReturn(Arrays.asList(beanDefinitionStringOne, beanDefinitionInteger));
+        when(beanScannerTwo.scan()).thenReturn(Collections.singletonList(beanDefinitionStringTwo));
+
+        when(beanDefinitionStringOne.name()).thenReturn("String");
+        doReturn(String.class).when(beanDefinitionStringOne).type();
+        when(beanDefinitionStringOne.instance()).thenReturn("String bean");
+
+        when(beanDefinitionInteger.name()).thenReturn("Integer");
+        doReturn(Integer.class).when(beanDefinitionInteger).type();
+        when(beanDefinitionInteger.instance()).thenReturn(2);
+
+        when(beanDefinitionStringTwo.name()).thenReturn("One more string bean");
+        doReturn(String.class).when(beanDefinitionStringTwo).type();
+        when(beanDefinitionStringTwo.instance()).thenReturn("String bean");
+
+        applicationContext = new ApplicationContextImpl(beanScannerList, validator, initializer);
+    }
+
+    @Test
+    @DisplayName("Application context initializing")
```

Comment:
Add in this name what we expect

Reply:
@DanielChanger changed to "Scan, validate and initialized methods have been called". Is that ok? If you have a better idea, please share it with me.